### PR TITLE
Article module fix read more title

### PR DESCRIPTION
**Pull request by RickR2H**
Pull Request for Issue #44135 

### Summary of Changes
Fix the read more title functionality.

### Testing Instructions
- Toggle the function Read More with Title on and off and see if the title is shown or hidden
- Change the Read More Limit (characters) to 15 to see if the title is shortened,

### Actual result BEFORE applying this Pull Request
Toggle Read More with Title not working

### Expected result AFTER applying this Pull Request
Toggle Read More with Title is working and truncation of title is working

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on b996b09473d4be5ce15fedec6bcfb7e13f3b66a9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44136">issues.joomla.org/tracker/joomla-cms/44136</a>.</sub>

**Comment by brianteeman** [comment]
https://github.com/user-attachments/assets/90957d3a-de69-4e97-98d1-d13f4e43ba1e

**Comment by RickR2H** [comment]
@brianteeman This option only reflects the settings in the module. Article parameters are not respected. This could be a feature we can add in the future. @drmenzelit What do you suggest?

**Comment by brianteeman** [comment]
My point is that the setting in the content component are preventing the setting in the module from working.

check it out yourself.
1. set the content component options to hide readmore title
2. set the module to show readmore title

expected behaviour - module shows the readmore title
actual behaviour - module does not show the readmore title

**Comment by RickR2H** [comment]
@brianteeman please test again.

File: modules/mod_articles/tmpl/default_items.php [hunk 1]
```diff
@@ -105,13 +107,19 @@
...
 
                         <?php if ($params->get('show_readmore')) : ?>
-                            <?php if ($params->get('show_readmore_title') === 0) : ?>
-                                <?php $item->params->set('show_readmore_title', 0); ?>
-                            <?php endif; ?>
-                            <?php if ($params->get('readmore_limit') >= 0) : ?>
-                                <?php $item->params->set('readmore_limit', $params->get('readmore_limit')); ?>
-                            <?php endif; ?>
-                            <?php echo LayoutHelper::render('joomla.content.readmore', ['item' => $item, 'params' => $item->params, 'link' => $item->link]); ?>
+                            <p class="mod-articles-category-readmore">
```
Comment (brianteeman) [inline modules/mod_articles/tmpl/default_items.php]: should this not be mod-articles-item-readmore ?

File: modules/mod_articles/tmpl/default_items.php [hunk 2]
```diff
@@ -105,13 +107,19 @@
...
+                                <a class="mod-articles-category-title btn btn-secondary" href="<?php echo $item->link; ?>">
```
Comment (brianteeman) [inline modules/mod_articles/tmpl/default_items.php]: should this not be mod-articles-item-title

**Comment by JeroenMoolenschot** [comment]
Thanks Rick!

One small inconsistency 

The module now shows > Read more ... Title.
image

While the component shows > Read more: Title
image

Without displaying the title it works both the same

**Comment by brianteeman** [comment]
Good spot
Without the title it should be using JGLOBAL_READ_MORE
With the title is should be using JGLOBAL_READ_MORE_TITLE

File: modules/mod_articles/tmpl/default_items.php (hunk 2 above)
Comment (RickR2H) [reply modules/mod_articles/tmpl/default_items.php]: Agreed. Thanks!

File: modules/mod_articles/tmpl/default_items.php (hunk 1 above)
Comment (RickR2H) [reply modules/mod_articles/tmpl/default_items.php]: Tnanks!

**Comment by RickR2H** [comment]
Thanks for the feedback! Hopefully fixed now.

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on e00e280cbe469df783545b43b9291643ee9643d3<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44136">issues.joomla.org/tracker/joomla-cms/44136</a>.</sub>

**Comment by JeroenMoolenschot** [comment]
I have tested this item :white_check_mark: successfully on e00e280cbe469df783545b43b9291643ee9643d3<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44136">issues.joomla.org/tracker/joomla-cms/44136</a>.</sub>

**Comment by RickR2H** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44136">issues.joomla.org/tracker/joomla-cms/44136</a>.</sub>

File: modules/mod_articles/tmpl/default_items.php [hunk 3]
```diff
@@ -18,17 +19,19 @@
...
         <?php
-            $displayInfo = $item->displayHits || $item->displayAuthorName || $item->displayCategoryTitle || $item->displayDate;
+        $displayInfo = $item->displayHits || $item->displayAuthorName || $item->displayCategoryTitle || $item->displayDate;
         ?>
         <li>
             <article class="mod-articles-item" itemscope itemtype="https://schema.org/Article">
 
                 <?php if ($params->get('item_title') || $displayInfo || $params->get('show_tags') || $params->get('show_introtext') || $params->get('show_readmore')) : ?>
-                    <div class="mod-articles-item-content">
+                    <div class="mod-articles-item-content<?php echo $item->active ? ' $item->active' : ''; ?>">
```
Comment (Quy) [inline modules/mod_articles/tmpl/default_items.php]: `$item->active` is enclosed in single quotes so it will output as it and not its value.

**Comment by brianteeman** [comment]
I think it would be better to do this the way that @rdeutz has done with #43457 - that way you can still use the layout etc

**Comment by brianteeman** [comment]
Removed by positive test as I think this is doing it the wrong way<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44136">issues.joomla.org/tracker/joomla-cms/44136</a>.</sub>

File: modules/mod_articles/tmpl/default_items.php (hunk 3 above)
Comment (RickR2H) [reply modules/mod_articles/tmpl/default_items.php]: Thanks! missed that!

**Comment by RickR2H** [comment]
@brianteeman please test the new version.

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on a81d79c2c72604fb6fe1a72371361648da1bb422<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44136">issues.joomla.org/tracker/joomla-cms/44136</a>.</sub>

File: modules/mod_articles/tmpl/default_items.php [hunk 4]
```diff
@@ -18,17 +19,19 @@
...
         <?php
-            $displayInfo = $item->displayHits || $item->displayAuthorName || $item->displayCategoryTitle || $item->displayDate;
+        $displayInfo = $item->displayHits || $item->displayAuthorName || $item->displayCategoryTitle || $item->displayDate;
         ?>
         <li>
             <article class="mod-articles-item" itemscope itemtype="https://schema.org/Article">
 
                 <?php if ($params->get('item_title') || $displayInfo || $params->get('show_tags') || $params->get('show_introtext') || $params->get('show_readmore')) : ?>
-                    <div class="mod-articles-item-content">
+                    <div class="mod-articles-item-content<?php echo $item->active ? ' active' : ''; ?>">
```
Comment (Hackwar) [inline modules/mod_articles/tmpl/default_items.php]: Where is this `active` class coming from?

Comment (RickR2H) [reply modules/mod_articles/tmpl/default_items.php]: In other modules there was an active class added to the active link. I added this so this functionally can be added with CSS if needed.

Comment (brianteeman) [reply modules/mod_articles/tmpl/default_items.php]: The question is under what condition will this ever be active

**Comment by richard67** [comment]
@RickR2H I've allowed myself to fix the merge conflict which was caused by your other, merged PR #44132 in file "modules/mod_articles/tmpl/default_items.php". What remains is the unresolved conversation above: https://github.com/joomla/joomla-cms/pull/44136/#discussion_r1779255235

**Comment by richard67** [comment]
Back to pending as the test by @JeroenMoolenschot has been invalidated by a later change which was not just a branch update. I will restore @brianteeman 's test result as that was just invalidated by a branch update (with a merge conflict resolved by me but that is not relevant for the test result). But the inresolved conversation should be resolved, and it needs a 2nd human test.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44136">issues.joomla.org/tracker/joomla-cms/44136</a>.</sub>

**Comment by brianteeman** [comment]
I have not tested this item.<br /><br />Removing my test as the active makes no sense <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44136">issues.joomla.org/tracker/joomla-cms/44136</a>.</sub>

File: modules/mod_articles/tmpl/default_items.php (hunk 4 above)
Comment (Hackwar) [reply modules/mod_articles/tmpl/default_items.php]: I've decided against this and removed it. We can add this as a new feature in 5.3.

File: modules/mod_articles/tmpl/default_items.php [hunk 5]
```diff
@@ -18,8 +19,10 @@
     $gridCols = 'grid-cols-' . $params->get('layout_columns');
 }
 
+// Get the language direction
+$direction = Factory::getApplication()->getLanguage()->isRtl() ? 'left' : 'right';
```
Comment (Quy) [inline modules/mod_articles/tmpl/default_items.php]: Remove this and the import statement.

Comment (RickR2H) [reply modules/mod_articles/tmpl/default_items.php]: I'll remove the obsolete code. Thanks!

**Comment by RickR2H** [comment]
@Quy could you please test again?

**Comment by Hackwar** [comment]
I merged this because the latest changes were removing the unrelated active status and the actual code in this PR was successfully tested before. Thank you @RickR2H for your contribution.

**Pull request merged**